src/iterators.rs

```diff
@@ -44,6 +45,52 @@ pub struct BFSSuccessors {
     pub index: usize,
 }
 
+#[pyproto]
+impl<'p> PyObjectProtocol<'p> for BFSSuccessors {
+    fn __richcmp__(
+        &self,
+        other: &'p PySequence,
+        op: pyo3::basic::CompareOp,
+    ) -> PyResult<bool> {
+        let compare = |other: &PySequence| -> PyResult<bool> {
+            if other.len()? as usize != self.bfs_successors.len() {
+                return Ok(false);
+            }
+            let gil = Python::acquire_gil();
+            let py = gil.python();
+            for i in 0..self.bfs_successors.len() {
+                let other_raw = other.get_item(i.try_into().unwrap())?;
+                let other_value: (PyObject, Vec<PyObject>) =
+                    other_raw.extract()?;
+                if self.bfs_successors[i].0.as_ref(py).compare(other_value.0)?
+                    != std::cmp::Ordering::Equal
+                {
+                    return Ok(false);
+                }
+                for (index, obj) in self.bfs_successors[i].1.iter().enumerate()
+                {
+                    if obj.as_ref(py).compare(&other_value.1[index])?
+                        != std::cmp::Ordering::Equal
```

Comment:
this is funny `if this is not equal to equal` (in a brain teasing way; I think it's good)

Reply:
Yeah, the api around comparisons is weird. It's because they're trying to map the python c api to rust's ordering https://docs.rs/pyo3/0.12.3/pyo3/struct.PyAny.html#method.compare it's unfortunately the easiest way to do 'a == b' for python objects. it would be nice if pyo3 had a `eq` method or something to do this and return a bool

Reply:
yeah, a bit roundabout.